luatest/server.lua

```diff
@@ -36,47 +49,109 @@ local Server = {
--- @string object.command Command to start server process.
--- @string object.workdir Value to be passed in `TARANTOOL_WORKDIR`.
--- @string[opt] object.chdir Path to cwd before running a process.
--- @tab[opt] object.env Table to pass as env variables to process.
--- @tab[opt] object.args Args to run command with.
--- @int[opt] object.http_port Value to be passed in `TARANTOOL_HTTP_PORT` and used to perform HTTP requests.
--- @int[opt] object.net_box_port Value to be passed in `TARANTOOL_LISTEN` and used for net_box connection.
--- @tab[opt] object.net_box_credentials Override default net_box credentials.
--- @string[opt] object.alias Instance alias. Used to prefix output.
--- @return input object.
+--- Build a server object.
+--
+-- @tab object
+-- @string[opt] object.command Executable path to run a server process with.
+--   Defaults to the internal `server_instance.lua` script. If a custom path
+--   is provided, it should correctly process all env variables listed below
+--   to make constructor parameters work.
+-- @tab[opt] object.args Arbitrary args to run `object.command` with.
+-- @tab[opt] object.env Pass the given env variables into the server process.
+-- @string[opt] object.chdir Change to the given directory before running
+--   the server process.
+-- @string[opt] object.alias Alias for the new server and the value of the
+--   `TARANTOOL_ALIAS` env variable which is passed into the server process.
+--   Defaults to 'server'.
+-- @string[opt] object.workdir Working directory for the new server and the
+--   value of the `TARANTOOL_WORKDIR` env variable which is passed into the
+--   server process.
+--   Defaults to `<vardir>/<alias>-<random id>`.
+-- @string[opt] object.datadir Directory path whose contents will be recursively
+--   copied into `object.workdir` during initialization.
+-- @number[opt] object.http_port Port for HTTP connection to the new server and
+--   the value of the `TARANTOOL_HTTP_PORT` env variable which is passed into
+--   the server process.
+--   Not supported in the default `server_instance.lua` script.
+-- @number[opt] object.net_box_port Port for the `net.box` connection to the new
+--   server and the value of the `TARANTOOL_LISTEN` env variable which is passed
+--   into the server process.
+-- @string[opt] object.net_box_uri URI for the `net.box` connection to the new
+--   server and the value of the `TARANTOOL_LISTEN` env variable which is passed
+--   into the server process.
+-- @tab[opt] object.net_box_credentials Override the default credentials for the
+--   `net.box` connection to the new server.
+-- @tab[opt] object.box_cfg Extra options for `box.cfg()` and the value of the
+--   `TARANTOOL_BOX_CFG` env variable which is passed into the server process.
+-- @return table
 function Server:new(object)
     checks('table', self.constructor_checks)
     self:inherit(object)
     object:initialize()
     return object
 end
 
+-- Initialize the server object.
 function Server:initialize()
```

Comment:
It's not a `local` function, but the docstring does not start with `---`.

Reply:
It is kind of an internal function. It shouldn't be invoked as `server:initialize()`, so I didn't document it.

BTW, the same should be for `server:build_env()`. This function is supposed to be an internal one as well.

Reply:
Then why aren't both functions `local`?

Reply:
Then 'inheritance' will not work correctly.